### ISSUE TITLE
MIM-Refiner on ConvNets

### ISSUE THREAD

**Issue by ricklisz** [ml-jku/MIM-Refiner#13]
Hi, thank you for your insightful work. I wonder have the authors tried applying MIM-Refiner on CNNs that are also pretrained in a MAE fashion (https://arxiv.org/abs/2301.03580)? If the model uses a hierarchical encoder-decoder for pretraining, do you think it can still benefit from MIM-refiner? Thanks!

**Comment by BenediktAlkin** [comment]
Hi, I have not tried out MIM-Refiner on CNNs but I dont see why the insights from ViTs should substantially differ from CNNs.

- Modern CNNs (since ConvNeXt) have a very similar block architecture, except they use (sparse) convolutions instead of attention to learn interactions between patches
- The bad off-the-shelf and few-shot performance mainly comes from the MIM pre-training task as it needs to reconstruct all pixels (or latent representations in methods like data2vec) of the image, hence it does not form an implicit object focus or transferable abstractions. So it should not matter what the architecture is, as long as the pre-training task is some kind of reconstruction objective.
- One thing to consider is that the benefit of multiple heads really kicks in for larger models (>=300M parameter models). I am not fully up to date what the largest MIM CNNs are, but from what I know they are often max ~200M parameters. For smaller models, the degrading representation is not that severe or absent. In this case you wouldn't need multiple heads and one head at the end of the encoder is sufficient. Nevertheless, refining the MIM model with a single instance discrimination head should greatly improve things like k-NN or few shot performance. For example, this is the setting that we use to refine B/16 models as they dont show degrading representation quality where MAE with ViT-B/16 improves from 51.1% k-NN to 76.6% k-NN (Table 21 in the appendix).
- I dont know exactly how many parameters/FLOPS the hierachical decoders have. I'd still expect them to be more lightweight than the encoder for fast training and because large decoders are not good for ViTs.

I would start by analyzing the k-NN accuracy at each layer of the CNN pre-trained with a MIM objective, like in Figure 3b of the paper. If it degrades towards the end refine the model with multiple instance discrimination heads, otherwise just stick to 1 head.

[here](https://github.com/ml-jku/MIM-Refiner/blob/main/src/yamls/analysis/knn_per_layer/maeog_b16.yaml) is the config yaml how you get the k-NN per block for these plots on a ViT-B. For the CNNs you would need to implement the architecture into the MIM-Refiner code. Alternatively, you can just extract the features from another codebase and use a simple k-NN classifier (e.g. [here](https://github.com/ml-jku/MIM-Refiner/blob/main/src/metrics/knn.py#L76) is the one we used for MIM-Refiner)

**Comment by ricklisz** [comment]
Nice, thank you so much for your quick reply!
